### ISSUE TITLE
Adding images and changing links for customization

### ISSUE THREAD

**Issue by jonsken** [GeoNodeUserGroup-DE/geonode-blueprint-docker#18]
First of all thank you for this amazing blueprint! To let you know I am pretty new to GeoNode and django - so if I am missing out basics, any links to improve my knowledge and get hands on are welcome.
I want to do some basic customization to the UI, like logo, backgrounds, links, etc.. I used parts of [this customization guide](https://training.geonode.geosolutionsgroup.com/master/GN4/101_CUSTOMIZE_LF.html#custom-header-and-footer) and created files in the following folders:
```
/geonode-project/geonode/geonode/apps/customizations/
|-- ...
|-- templates/
|    |-- ...
|    +-- geonode-mapstore-client/
|    |    |-- ...
|    |    +-- snippets/
|    |         |-- ...
|    |         |-- brand_navbar.html
|    |         +-- footer.html
|-- ...
```
The added files are read and the navbar and footer are loaded. I also wanted to custom images (like logo, header, ...) and if I inspect the logo in the navbar, the link says `/static/mapstore/img/geonode-logo.svg`, but I am not finding a `static`-folder in my folder tree in order to add custom images. Where do I add/change images in order to appear/be loaded in the UI?
For adding/changing links between sites is there some kind of file or link tree available somewhere for making linking between sites easier?

**Comment by kikislater** [comment]
From docs:
```
/opt/geonode-project/my_geonode/src/my_geonode/
|-- ...
|-- static/
|    |-- ...
|    +-- img/
|         |-- ...
|         |-- hero.jpeg
|         +-- logo.png
|-- ...
```

So you should have the same structure but for static files. Then you have to run django admin command to have static files available: https://docs.djangoproject.com/en/4.2/howto/static-files/

**Comment by jonsken** [comment]
When setting up GeoNode like in the GeoNode docs (e.g. [like this](https://docs.geonode.org/en/master/install/basic/index.html#geonode-basic-installation)), I do have the `static` folder and files, but then GeoNode is not properly running. 
When using the blueprint, there is nothing like this to be found and the overall folder structure also changes:
![grafik](https://github.com/user-attachments/assets/43532e02-95b5-4dc0-a2fb-246770ee239e)


As far as I understand there needs `$ python manage.py collectstatic` to be run, but I also can not find `manage.py` anywhere...

**Comment by kikislater** [comment]
But I don't know why you are looking this custom geonode project whereas you don't understand geonode ... 
Where are static files is not difficult to find in this project (In settings and docker you will find it!). If you would like to understand I suggest using geonode-project first instead of this one !

**Comment by jonsken** [comment]
I would have love to use the [basic guide](https://docs.geonode.org/en/master/install/basic/index.html#geonode-basic-installation) or the [advanced guide](https://docs.geonode.org/en/master/install/advanced/index.html) in order to set up GeoNode and learn it through the GeoNode tutorials, but neither of them worked (conflicted dependencies, incomplete project files/folder due to problems during installation/guide, etc.). The GeoNode blueprint was the first set up, which worked to set up and in the end also worked during running the servers, that is why I tried to get a hand on it

**Comment by ridoo** [comment]
@jonsken good to see this setup helped you to start up GeoNode.

However, you should keep in mind that this blueprint is meant for long-term maintenance of projects using GeoNode, i.e. feature development and/or GeoNode project customizations -- similar to what the geonode-project provides. The main difference is that the actual setup is somewhat opinionated [as described in the README](https://github.com/GeoNodeUserGroup-DE/geonode-blueprint-docker#background) to allow more flexibility and better tracking the upstream.

I do not know what errors you got when starting with the basic/advanced guides. But I am sure, if you do not understand that errors and how to resolve them, you would probably run into troubles fast when using this setup as the same software is being used but packaged differently (for reasons). You have to know the basics of Docker (compose), Django, decontainers and of course of GeoNode itself. 

To answer your concrete question regarding the static files:

> The added files are read and the navbar and footer are loaded. I also wanted to custom images (like logo, header, ...) and if I inspect the logo in the navbar, the link says /static/mapstore/img/geonode-logo.svg, but I am not finding a static-folder in my folder tree in order to add custom images. Where do I add/change images in order to appear/be loaded in the UI?
For adding/changing links between sites is there some kind of file or link tree available somewhere for making linking between sites easier?

**tl;dr**: The file `/static/mapstore/img/geonode-logo.svg` is [part of the geonode-mapstore-client](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/static/mapstore/img/geonode-logo.svg) which is packaged as django contribe module. The logo is underneath the `static` folder which lets Django finds it as a static resource. During the build process all static files are copied to be served by nginx.


This project is not meant to just replacing files. The aim is to stay as close to the upstream as possible to avoid conflicts when following upstream over time. We try to approach this by using Docker and Git features like customizing base images, volume overlays, or submodules. However, as the whole tree structure is packaged in the Docker image built you can "see" it either when running `docker compose exec -it <container name> bash` or when starting the setup via devcontainers. Doing so would require you to know where non-volatile changes can be made as containers are considered to be throw-away artifacts.
